src/ext/java/org/opentripplanner/ext/legacygraphqlapi/LegacyGraphQLAPI.java

```diff
@@ -0,0 +1,189 @@
+
+  public LegacyGraphQLAPI(
+      @Context OTPServer otpServer,
+      @Context Providers providers,
+      @PathParam("routerId") String routerId
+  ) {
+    this.router = otpServer.getRouter();
+
+    ContextResolver<ObjectMapper> resolver =
+        providers.getContextResolver(ObjectMapper.class, MediaType.APPLICATION_JSON_TYPE);
+    ObjectMapper mapper = resolver.getContext(ObjectMapper.class);
+    mapper.setDefaultPropertyInclusion(JsonInclude.Include.ALWAYS);
+  }
+
+  @POST
+  @Path("/")
+  @Consumes(MediaType.APPLICATION_JSON)
+  public Response getGraphQL(
+      HashMap<String, Object> queryParameters,
+      @HeaderParam("OTPMaxResolves") @DefaultValue("1000000") int maxResolves,
+      @Context HttpHeaders headers
+  ) {
+    if (queryParameters == null || !queryParameters.containsKey("query")) {
+      LOG.debug("No query found in body");
+      return Response
+          .status(Response.Status.BAD_REQUEST)
+          .type(MediaType.TEXT_PLAIN_TYPE)
+          .entity("No query found in body")
+          .build();
+    }
+
+    Locale locale = headers.getAcceptableLanguages().size() > 0
+        ? headers.getAcceptableLanguages().get(0)
+        : router.defaultRoutingRequest.locale;
+
+    String query = (String) queryParameters.get("query");
+    Object queryVariables = queryParameters.getOrDefault("variables", null);
+    String operationName = (String) queryParameters.getOrDefault("operationName", null);
+    Map<String, Object> variables;
+    if (queryVariables instanceof Map) {
+      variables = (Map) queryVariables;
+    }
+    else if (queryVariables instanceof String && !((String) queryVariables).isEmpty()) {
+      try {
+        variables = deserializer.readValue((String) queryVariables, Map.class);
+      }
+      catch (IOException e) {
+        return Response
+            .status(Response.Status.BAD_REQUEST)
+            .type(MediaType.TEXT_PLAIN_TYPE)
+            .entity("Variables must be a valid json object")
+            .build();
+      }
+    }
+    else {
+      variables = new HashMap<>();
+    }
+    return LegacyGraphQLIndex.getGraphQLResponse(
+        query,
+        router,
+        variables,
+        operationName,
+        maxResolves,
+        locale
+    );
+  }
+
+  @POST
+  @Path("/")
+  @Consumes("application/graphql")
+  public Response getGraphQL(
+      String query,
+      @HeaderParam("OTPMaxResolves") @DefaultValue("1000000") int maxResolves,
+      @Context HttpHeaders headers
+  ) {
+    Locale locale = headers.getAcceptableLanguages().size() > 0
+        ? headers.getAcceptableLanguages().get(0)
+        : router.defaultRoutingRequest.locale;
+    return LegacyGraphQLIndex.getGraphQLResponse(query, router, null, null, maxResolves, locale);
+  }
+
+  @POST
+  @Path("/batch")
+  @Consumes(MediaType.APPLICATION_JSON)
+  public Response getGraphQLBatch(
+      List<HashMap<String, Object>> queries,
+      @HeaderParam("OTPTimeout") @DefaultValue("10000") int timeout,
```

Comment:
In the current hsldevcom version the default value is 30000

Reply:
done